task/task.go

```diff
@@ -29,7 +39,9 @@ func DoRetryWithTimeout(t func() (interface{}, bool, error), timeout, timeBefore
 				}
 
 			default:
+				errLock.Lock()
 				out, retry, err = t()
+				errLock.Unlock()
 				if err == nil || !retry {
 					done <- true
```

Comment:
errLock.Unlock() should be done after sending `true` over the done channel.

Reply:
fixed